cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -168,23 +178,19 @@ public String commitAndPush(CommitMeta commitMeta, List<Path> filesChanged)
       throw new NoChangesToPushException("No changes have been maid for the applications to push.");
     }
 
-    for (Path path : filesChanged) {
-      git.add().addFilepattern(path.toString()).call();
-    }
+    git.add().addFilepattern(fileChanged.toString()).call();
 
     RevCommit commit = getCommitCommand(commitMeta).call();
 
-    // Fail if the status is not clean i.e. there are unexpected changes.
-    Status postCommitStatus = git.status().call();
-    if (!postCommitStatus.isClean()) {
-      throw new UnexpectedRepositoryChangesException(String.format("Unexpected changes other than changed files %s",
-                                                                   filesChanged));
-    }
-
     PushCommand pushCommand = createCommand(git::push, sourceControlConfig, credentialsProvider);
     pushCommand.call();
 
-    return commit.getId().getName();
+    try {
+      TreeWalk walk = TreeWalk.forPath(git.getRepository(), fileChanged.toString(), commit.getTree());
+      return walk.getObjectId(0).getName();
+    } catch (IOException e) {
+      throw new PushFailureException(String.format("Failed to parse fileHash for application %s", fileChanged));
```

Comment:
parse -> get

Reply:
I don't think this should result in a failure, since we successfully pushed the commit. This seems more like a warning log that results in an empty file hash.

Reply:
yeah, I've changed it to log warning and return null fileHash